newrelic/hooks/mlmodel_sklearn.py

```diff
@@ -283,15 +235,38 @@ def bind_predict(X, *args, **kwargs):
     return X
 
 
-def create_feature_event(transaction, class_, inference_id, instance, args, kwargs):
+def create_prediction_event(transaction, class_, inference_id, instance, args, kwargs, return_val):
     import numpy as np
 
     data_set = bind_predict(*args, **kwargs)
     model_name = getattr(instance, "_nr_wrapped_name", class_)
     model_version = getattr(instance, "_nr_wrapped_version", "0.0.0")
     user_provided_feature_names = getattr(instance, "_nr_wrapped_feature_names", None)
+    label_names = getattr(instance, "_nr_wrapped_label_names", None)
     settings = transaction.settings if transaction.settings is not None else global_settings()
 
+    labels = []
+    if return_val is not None:
+        if not hasattr(return_val, "__iter__"):
+            labels = np.array([return_val])
```

Comment:
Have we tested that this works for string return vals? Seems like it might split something like `"abc"` to `["a", "b", "c"]`.

Reply:
Very good point! For the label columns, the values are only numeric (these values come from the output of the model prediction) so we should not need to handle this scenario.